native/common/include/jp_class.h

```diff
@@ -192,10 +186,10 @@ class JPClass : public JPResource
 
 	JPContext* getContext() const
 	{
+		// This sanity check is for during shutdown.
 		if (m_Context == 0)
-			JP_RAISE(PyExc_RuntimeError, "Null context");
-		return
-		m_Context;
+			JP_RAISE(PyExc_RuntimeError, "Null context"); // GCOVR_EXCL_LINE
```

Comment:
is this the exception you are catching in the atexit handler? If so, I'd prefer a custom exception type (or if you object the overhead, leave it like this).

Reply:
Null context should be a SystemError rather than RuntimeError.  There is a screen at the PyJP level to prevent ever getting to the backend when the JVM is not running.  I have been working to correct that issue.